apps/widclkinfo/widget.js

```diff
@@ -1,50 +1,93 @@
-if (!require("clock_info").loadCount) { // don't load if a clock_info was already loaded
-  // Load the clock infos
-  let clockInfoItems = require("clock_info").load();
-  // Add the
-  let clockInfoMenu = require("clock_info").addInteractive(clockInfoItems, {
-    app : "widclkinfo",
-    // Add the dimensions we're rendering to here - these are used to detect taps on the clock info area
-    x : 0, y: 0, w: 72, h:24,
-    // You can add other information here you want to be passed into 'options' in 'draw'
-    // This function draws the info
-    draw : (itm, info, options) => {
-      // itm: the item containing name/hasRange/etc
-      // info: data returned from itm.get() containing text/img/etc
-      // options: options passed into addInteractive
-      clockInfoInfo = info;
-      if (WIDGETS["clkinfo"])
-        WIDGETS["clkinfo"].draw(WIDGETS["clkinfo"]);
-    }
-  });
-  let clockInfoInfo; // when clockInfoMenu.draw is called we set this up
+(() => {
+  if (!require("clock_info").loadCount) { // don't load if a clock_info was already loaded
+  const clock_info = require("clock_info");
 
-  // The actual widget we're displaying
   WIDGETS["clkinfo"] = {
-    area:"tl",
-    width: clockInfoMenu.w,
-    draw:function(e) {
-      clockInfoMenu.x = e.x;
-      clockInfoMenu.y = e.y;
-      var o = clockInfoMenu;
-      // Clear the background
+    area: "tl",
+    width: 0, //this.clockInfoMenu.w,
+    init: function() {
+      this.width = this.clockInfoMenu.w;
+      delete this.init;
+      return this;
+    },
+    clockInfoInfo: undefined, // defined during clockInfoMenu.draw()
```

Comment:
Perhaps we give this a less repeating name

```suggestion
    info: undefined, // defined during clockInfoMenu.draw()
```